src/MongoDB.Driver/Linq/Processors/OutOfCurrentScopePrefixCollector.cs

```diff
@@ -0,0 +1,53 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using MongoDB.Driver.Linq.Expressions;
+
+namespace MongoDB.Driver.Linq.Processors
+{
+    /// <summary>
+    /// The expression visitor to collect all expression fields which have been defined out of the current scope.
+    /// </summary>
+    internal sealed class OutOfCurrentScopePrefixCollector : ExtensionExpressionVisitor
+    {
+        private readonly List<IHasOutOfCurrentScopePrefix> _outOfCurrentScopePrefixCollection = new List<IHasOutOfCurrentScopePrefix>();
+
+        /// <summary>
+        /// Collects the list of node information.
+        /// </summary>
+        /// <param name="expression">The expression.</param>
+        /// <returns>The result collection.</returns>
+        public static IEnumerable<IHasOutOfCurrentScopePrefix> Collect(Expression expression)
+        {
+            var parameterCollector = new OutOfCurrentScopePrefixCollector();
+            parameterCollector.Visit(expression);
+            return parameterCollector._outOfCurrentScopePrefixCollection.Distinct();
+        }
+
+        public override Expression Visit(Expression node)
+        {
+            var memberInfo = node as IHasOutOfCurrentScopePrefix;
```

Comment:
memberInfo doesn't make much sense as the variable name here.

Reply:
Done